types/Projects.ts

```diff
@@ -49,3 +49,74 @@ export type ProjectTemplateRepoConfig = {
   projects?: ProjectTemplate[];
   components?: ComponentTemplate[];
 };
+
+export type PrivateAppComponentConfig = {
+  name: string;
+  description: string;
+  uid: string;
+  scopes: Array<string>;
+  public: boolean;
+  extensions?: {
+    crm: {
+      cards: Array<{ file: string }>;
+    };
+  };
+};
+
+export type PublicAppComponentConfig = {
+  name: string;
+  uid: string;
+  description: string;
+  allowedUrls: Array<string>;
+  auth: {
+    redirectUrls: Array<string>;
+    requiredScopes: Array<string>;
+    optionalScopes: Array<string>;
+    conditionallyRequiredScopes: Array<string>;
+  };
+  support: {
+    supportEmail: string;
+    documentationUrl: string;
+    supportUrl: string;
+    supportPhone: string;
+  };
+  extensions?: {
+    crm: {
+      cards: Array<{ file: string }>;
+    };
+  };
+  webhooks?: {
+    file: string;
+  };
+};
+
+export type AppCardComponentConfig = {
+  type: 'crm-card';
+  data: {
+    title: string;
+    uid: string;
+    location: string;
+    module: {
+      file: string;
+    };
+    objectTypes: Array<{ name: string }>;
+  };
+};
+
+export type GenericComponentConfig =
+  | PublicAppComponentConfig
+  | PrivateAppComponentConfig
+  | AppCardComponentConfig;
+
+export enum ComponentTypes {
```

Comment:
I moved all of these to the types/projects file, and I switched `ComponentTypes` to be an enum. It felt redundant to have both `ComponentTypes` and `COMPONENT_TYPES`. This let me replace `COMPONENT_TYPES` everywhere in favor of the `ComponentTypes` enum